src/integrity-checker.js

```diff
@@ -292,16 +300,19 @@ export default class InstallationIntegrityChecker {
         return 'FILES_MISSING';
       }
 
-      // Since we know the "files" entry is sorted alphabetically, we can optimize the thing
+      // Since we know the "files" entry is sorted (alphabetically), we can optimize the thing
       // Instead of storing the files in a Set, we can just iterate both arrays at once. O(n)!
       for (let u = 0, v = 0; u < expected.files.length; ++u) {
-        // Skip over files that have been added
-        while (v < actual.files.length && actual.files[v] !== expected.files[u]) {
+        // Number of iterations after which there won't be enough entries remaining for the arrays to match
+        const max = actual.files.length - expected.files.length;
```

Comment:
I did this originally but I find it more readable for this variable and its comment to be close to the place we're using it. Since this is something that could/should be optimized by a decent compiler I think it's fine.